### ISSUE TITLE
Tooltip/popover cannot be initialized right after it is destroyed

### ISSUE THREAD

**Issue by gxoptg** [twbs/bootstrap#16230]
The following code:

```js
$(".tooltip")
  .tooltip("destroy")
  .tooltip( { title: "A title" } );
```

destroys the tooltip, but does not initialize the new one.

In fact, the bug is caused because the tooltip is not destroyed immediately after calling [`Tooltip.prototype.destroy`](https://github.com/twbs/bootstrap/blob/master/js/tooltip.js#L438), but after the jQuery’s `hide()` animation was finished:

```js
Tooltip.prototype.destroy = function () {
  var that = this
  clearTimeout(this.timeout)
  this.hide(function () { // ← Here
    that.$element.off('.' + that.type).removeData('bs.' + that.type)
  })
}
```

So, when `.tooltip()` is called right after `.tooltip("destroy")`, the tooltip data is present yet, and no action is performed:

```js
function Plugin(option) {
  return this.each(function () {
    var $this   = $(this)
    var data    = $this.data('bs.tooltip')
    var options = typeof option == 'object' && option

    if (!data && /destroy|hide/.test(option)) return
    // ↓ Here, `data` is still defined
    if (!data) $this.data('bs.tooltip', (data = new Tooltip(this, options)))
    if (typeof option == 'string') data[option]()
  })
}
```

The bug was created in the commit https://github.com/twbs/bootstrap/commit/76762169c359b381f70fa518daef09e76b0aefe3.

Working demo: http://jsbin.com/fepala/4/edit?html,js,output.

**Comment by twbs-lmvtfy** [comment]
Hi @gxoptg!

You appear to have posted a live example (http://jsbin.com/fepala/4/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness
* line 11, column 5: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 12, column 3: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
I would say that this needs a "destroyed" event, that you would then listen for prior to performing the reinitialization. Although I forget whether or not we've made a decision regarding adding "destroyed" events to Bootstrap v3.

**Comment by china0622** [comment]
Destory

**Comment by peterblazejewicz** [comment]
> I would say that this needs a "destroyed" event, that you would then listen for prior to performing the reinitialization. Although I forget whether or not we've made a decision regarding adding "destroyed" events to Bootstrap v3.

I've read the other issue on this:
https://github.com/twbs/bootstrap/issues/15607

**Comment by cvrebert** [comment]
Thanks!
Closing in favor of #15607 then.
